### ISSUE TITLE
Split thought duplicates lexeme

### ISSUE THREAD

**Issue by raineorshine** [cybersemics/em#1559]
## Steps to Reproduce

1. Create the thought `*test`
2. Place the caret after `*` and hit Enter to split the thought into `*` and `test`

## Current Behavior

`test` has two lexeme contexts (indicated by the superscript '2')

## Expected Behavior

`test` should only have one lexeme context (no superscript)

Lexeme contexts are stored in `state.thoughts.lexemeIndex[...].contexts`

**Comment by karunkop** [comment]
## Root cause of the issue

When we are hashing lexeme values, `hashThought` uses `normalizeThoughts` which trims a string in such a way that it replaces non word characters i.e `\W`. So what's happenning is there is being a duplication in `hashValue` since when we split `*test` as `*` and `test`, same hash value for test is being created and since they will be in different contexts, `text` lexeme is getting two context values in array.

**Comment by raineorshine** [comment]
The solution in #1563 relies on a side effect in `deleteData` in which the `Lexeme` is deleted directly from `IndexedDB`. Reducers should be pure functions, but it did not seem to cause any problems so it was accepted as an imperfect but effective solution.

Unfortunately, it ended up causing background errors when running tests, as `IndexedDB` does not exist in the NodeJS environment. In order to clean up the tests, I investigated an alternative solution that prevents the duplicate `Lexeme` at a lower level.

When `mergeConflictingLexemeIndexUpdates` finds pulled Lexeme contexts that are not included in the batch, it adds them. This ensures that additional Lexeme contexts from other devices are added. However, it fails to recognize when these other contexts have been edited, and thus no longer belong to the new Lexeme. For example, in the test case for this issue, `*test` is edited to `*`. The new thought no longer has the same `Lexeme` as `test`, but `mergeConflictingLexemeIndexUpdates` was not set up to detect this.

In 8e6f761 I have added a check in `mergeConflictingLexemeIndexUpdates` that makes sure the thought value has not changed, and  ensures the invalid Lexeme context is never added.
